docs/source/learn/examples/dimensionality.ipynb

```diff
@@ -0,0 +1,742 @@
+{
```

Comment:
<p><code>mu=[0, 0]</code></p><p> </p><p>Technically, it's incorrect to specify a scalar mu for the multivariate normal, although for backwards compatibility we reshape mu behind the scenes</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5437/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Reply:
If its just for backwards compability why dont we drop it?

Reply:
We can drop things after warning about them for long enough. If you want to drop supporting the automatic broadcast, then we'll need FutureWarning in a release version for about a year. After that we can make the break and argue that it's okay to not bump the major version.

Reply:
But we have a major release coming up where breaks are expected.

Reply:
We can drop it with a useful ValueError about the changed behavior

Reply:
I agree with Thomas, we're releasing a major version this is *our chance* to reform the code without the need for FutureWarning.

This is good for both devs and the community. For example in this case some of these incorrect but sort of works situations makes using the library more challenging. As a dev when I try and figure out what's happening I have to navigate the maze of conditionals and reference calls in the backend to figure out what is happening. I also need to read through so many much more code and tests to make a useful contribution.

In the nicest way I'm finding Ricardo and Michael are PyMC power users and they're not in the maze. You guys are so smart you have this elevated view and can see this maze from above, avoid all the traps, and know all the shortcuts. For noobs like me I run into all the deadends and then have to backtrack and try another route.

My strong pitch is we make the maze simpler. As with all codebases the complexity will increase over time, look at Theano. But we have this one opportunity right to make things simpler so please lets please do it, if not for you for me :heart: